src/metabase/api/dashboard.clj

```diff
@@ -405,6 +406,30 @@
                                                     (dissoc :id :entity_id :created_at :updated_at))))]
     (zipmap (map :id existing-tabs) new-tab-ids)))
 
+(defn- update-colvalmap-setting
+  "Visualizer dashcards have unique visualization settings which embed column id remapping metadata
+  This function iterates through the `:columnValueMapping` viz setting and updates referenced card ids
+
+  col->val-source can look like:
+  {:COLUMN_2 [{:sourceId 'card:<OLD_CARD_ID>', :originalName 'sum', :name 'COLUMN_2'}], ...}"
+  [col->val-source id->new-card]
+  (let [update-cvm-item (fn [item]
+                          (if-let [source-id (:sourceId item)]
+                            (if (and (string? source-id) (str/starts-with? source-id "card:"))
+                              (let [old-id (Long/parseLong (subs source-id 5))]
+                                (if-let [new-card (and old-id (get id->new-card old-id))]
+                                  (assoc item :sourceId (str "card:" (:id new-card)))
+                                  item))
+                              item)
+                            item))
```

Comment:
I think it'd be a bit more readable to use a regex here with a capturing group rather than `starts-with?` and `subs`. Something like `(re-find #"^card:(\d+)$" source-id)`